extensions/firmware/org.eclipse.smarthome.firmware.filesystem.test/src/test/groovy/org/eclipse/smarthome/firmware/filesystem/FileSystemFirmwareProviderTest.groovy

```diff
@@ -0,0 +1,724 @@
+        file << "prerequisiteVersion = " + V111 + "\n"
+        file << "changelog = " + CHANGELOG_EN + "\n"
+        file << "onlineChangelog = " + CHANGELOG_URL + "\n"
+        file << "md5hash = " + HASH_112 + "\n"
+
+        file = new File(THING_TYPE_1_DIR.getPath(), V112 + "_de_DE.properties")
+        file << "description = " + DESCRIPTION_DE + "\n"
+        file << "prerequisiteVersion = " + V111 + "\n"
+        file << "changelog = " + CHANGELOG_DE + "\n"
+        file << "onlineChangelog = " + CHANGELOG_URL + "\n"
+
+        file = new File(THING_TYPE_2_DIR.getPath(), IMAGE_V111)
+        file << IMAGE_V111;
+
+        file = new File(THING_TYPE_2_DIR.getPath(), V111 + ".properties")
+        file << "version = " + VERSION_111 + "\n"
+        file << "image = " + IMAGE_V111 + "\n"
+        file << "model = " + MODEL_B + "\n"
+        file << "vendor = " + VENDOR + "\n"
+        file << "description = " + DESCRIPTION_EN + "\n"
+        file << "changelog = " + CHANGELOG_EN + "\n"
+        file << "onlineChangelog = " + CHANGELOG_URL + "\n"
+        file << "md5hash = " + HASH_111 + "\n"
+
+        file = new File(THING_TYPE_2_DIR.getPath(), V111 + "_de_DE.properties")
+        file << "description = " + DESCRIPTION_DE + "\n"
+        file << "changelog = " + CHANGELOG_DE + "\n"
+        file << "onlineChangelog = " + CHANGELOG_URL + "\n"
+
+        file = new File(THING_TYPE_2_DIR.getPath(), IMAGE_V112)
+        file << IMAGE_V112;
+
+        file = new File(THING_TYPE_2_DIR.getPath(), V112 + ".properties")
+        file << "version = " + VERSION_112 + "\n"
+        file << "image = " + IMAGE_V112 + "\n"
+        file << "model = " + MODEL_B + "\n"
+        file << "vendor = " + VENDOR + "\n"
+        file << "description = " + DESCRIPTION_EN + "\n"
+        file << "prerequisiteVersion = " + V111 + "\n"
+        file << "changelog = " + CHANGELOG_EN + "\n"
+        file << "onlineChangelog = " + CHANGELOG_URL + "\n"
+        file << "md5hash = " + HASH_112 + "\n"
+
+        file = new File(THING_TYPE_2_DIR.getPath(), V112 + "_de_DE.properties")
+        file << "description = " + DESCRIPTION_DE + "\n"
+        file << "prerequisiteVersion = " + V111 + "\n"
+        file << "changelog = " + CHANGELOG_DE + "\n"
+        file << "onlineChangelog = " + CHANGELOG_URL + "\n"
+
+        System.setProperty(ConfigConstants.USERDATA_DIR_PROG_ARGUMENT, USER_DIR)
+        PROVIDER.activate()
+    }
+
+    @AfterClass
+    static void tearDownAfterClass() {
+        PROVIDER.deactivate()
+        def userRoot = new File(USER_DIR)
+        userRoot.deleteDir()
+        def root = new File(ROOT_DIR)
+    }
+
+    @Before
+    void setup() {
+        defaultLocale = Locale.getDefault()
+        Locale.setDefault(Locale.ENGLISH)
+    }
+
+    @After
+    void tearDown() {
+        Locale.setDefault(defaultLocale)
+    }
+
+    @Test
+    void 'get firmware for thing type 1 with version 111 with german locale'() {

```

Comment:
Groovy tests should no longer be used. So please convert all Groovy tests to Java tests 

Additionally:
* Remove the Groovy imports from the manifest
* Remove the Groovy nature from the Eclipse project